# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1365792711 14400
#      Fri Apr 12 14:51:51 2013 -0400
# Node ID ff061068a66c24365c87f38db7a7eeb238ae0710
# Parent  49b059bf27c73d99f7102805c7cc59c41a7059c8
move dialog files to separate directory

* scripts/ui/errordlg.m, scripts/ui/helpdlg.m, scripts/ui/inputdlg.m,
scripts/ui/listdlg.m, scripts/ui/msgbox.m, scripts/ui/questdlg.m,
scripts/ui/warndlg.m: Move here from scripts/java.
* scripts/java/module.mk (java_FCN_FILES): Update list.
* scripts/ui/module.mk: New file.
* scripts/Makefile.am: Include it.
(ui/PKG_ADD, $(ui_GEN_FCN_FILES), ui/$(octave_dirstamp)): New targets.

diff --git a/scripts/Makefile.am b/scripts/Makefile.am
--- a/scripts/Makefile.am
+++ b/scripts/Makefile.am
@@ -69,16 +69,17 @@ include special-matrix/module.mk
 include startup/module.mk
 include statistics/base/module.mk
 include statistics/distributions/module.mk
 include statistics/models/module.mk
 include statistics/tests/module.mk
 include strings/module.mk
 include testfun/module.mk
 include time/module.mk
+include ui/module.mk
 
 nobase_fcnfile_DATA = $(FCN_FILES) $(GEN_FCN_FILES) $(JAR_FILES)
 
 image_DATA = $(IMAGES)
 
 FCN_FILES_IN = $(GEN_FCN_FILES:.m=.in)
 
 if AMCOND_BUILD_DOCS
@@ -208,16 +209,20 @@ strings/PKG_ADD: $(strings_FCN_FILES) $(
 testfun/PKG_ADD: $(testfun_FCN_FILES) $(testfun_GEN_FCN_FILES) testfun/$(octave_dirstamp) mk-pkg-add
 	$(srcdir)/mk-pkg-add $(srcdir) $(testfun_FCN_FILES) -- $(testfun_GEN_FCN_FILES) > $@-t
 	mv $@-t $@
 
 time/PKG_ADD: $(time_FCN_FILES) $(time_GEN_FCN_FILES) time/$(octave_dirstamp) mk-pkg-add
 	$(srcdir)/mk-pkg-add $(srcdir) $(time_FCN_FILES) -- $(time_GEN_FCN_FILES) > $@-t
 	mv $@-t $@
 
+ui/PKG_ADD: $(ui_FCN_FILES) $(ui_GEN_FCN_FILES) ui/$(octave_dirstamp) mk-pkg-add
+	$(srcdir)/mk-pkg-add $(srcdir) $(ui_FCN_FILES) -- $(ui_GEN_FCN_FILES) > $@-t
+	mv $@-t $@
+
 $(@ftp_GEN_FCN_FILES): @ftp/$(octave_dirstamp)
 $(audio_GEN_FCN_FILES): audio/$(octave_dirstamp)
 $(deprecated_GEN_FCN_FILES): deprecated/$(octave_dirstamp)
 $(elfun_GEN_FCN_FILES): elfun/$(octave_dirstamp)
 $(general_GEN_FCN_FILES): general/$(octave_dirstamp)
 $(geometry_GEN_FCN_FILES): geometry/$(octave_dirstamp)
 $(help_GEN_FCN_FILES): help/$(octave_dirstamp)
 $(image_GEN_FCN_FILES): image/$(octave_dirstamp)
@@ -239,16 +244,17 @@ time/PKG_ADD: $(time_FCN_FILES) $(time_G
 $(startup_GEN_FCN_FILES): startup/$(octave_dirstamp)
 $(statistics_base_GEN_FCN_FILES): statistics/base/$(octave_dirstamp)
 $(statistics_distributions_GEN_FCN_FILES): statistics/distributions/$(octave_dirstamp)
 $(statistics_models_GEN_FCN_FILES): statistics/models/$(octave_dirstamp)
 $(statistics_tests_GEN_FCN_FILES): statistics/tests/$(octave_dirstamp)
 $(strings_GEN_FCN_FILES): strings/$(octave_dirstamp)
 $(testfun_GEN_FCN_FILES): testfun/$(octave_dirstamp)
 $(time_GEN_FCN_FILES): time/$(octave_dirstamp)
+$(ui_GEN_FCN_FILES): ui/$(octave_dirstamp)
 
 @ftp/$(octave_dirstamp):
 	$(MKDIR_P) @ftp
 	: > @ftp/$(octave_dirstamp)
 audio/$(octave_dirstamp):
 	$(MKDIR_P) audio
 	: > audio/$(octave_dirstamp)
 deprecated/$(octave_dirstamp):
@@ -333,16 +339,19 @@ strings/$(octave_dirstamp):
 	$(MKDIR_P) strings
 	: > strings/$(octave_dirstamp)
 testfun/$(octave_dirstamp):
 	$(MKDIR_P) testfun
 	: > testfun/$(octave_dirstamp)
 time/$(octave_dirstamp):
 	$(MKDIR_P) time
 	: > time/$(octave_dirstamp)
+ui/$(octave_dirstamp):
+	$(MKDIR_P) ui
+	: > ui/$(octave_dirstamp)
 
 if AMCOND_BUILD_DOCS
 
 .DOCSTRINGS: $(FCN_FILES) $(GEN_FCN_FILES) mkdoc.pl Makefile
 	if [ "x$(srcdir)" != "x." ] && [ -f $(srcdir)/DOCSTRINGS ] && [ ! -f DOCSTRINGS ]; then \
 		cp $(srcdir)/DOCSTRINGS DOCSTRINGS; \
 		touch -r $(srcdir)/DOCSTRINGS DOCSTRINGS; \
 	fi
@@ -459,9 +468,8 @@ DISTCLEANFILES += \
   $(PKG_ADD_FILES) \
   $(DIRSTAMP_FILES) \
   $(GEN_FCN_FILES)
 
 distclean-local:
 	if [ "x$(srcdir)" != "x." ]; then \
 	  rm $(java_JAVA_IMAGES); \
 	fi
-	
diff --git a/scripts/java/module.mk b/scripts/java/module.mk
--- a/scripts/java/module.mk
+++ b/scripts/java/module.mk
@@ -1,25 +1,18 @@
 FCN_FILE_DIRS += java
 
 java_FCN_FILES = \
   java/dlgtest.m \
-  java/errordlg.m \
-  java/helpdlg.m \
-  java/inputdlg.m \
   java/javaArray.m \
   java/javaaddpath.m \
   java/javaclasspath.m \
   java/javamem.m \
   java/javarmpath.m \
-  java/listdlg.m \
-  java/msgbox.m \
-  java/questdlg.m \
-  java/usejava.m \
-  java/warndlg.m
+  java/usejava.m
 
 FCN_FILES += $(java_FCN_FILES)
 
 PKG_ADD_FILES += java/PKG_ADD
 
 DIRSTAMP_FILES += java/$(octave_dirstamp)
 
 if AMCOND_HAVE_JAVA
diff --git a/scripts/java/errordlg.m b/scripts/ui/errordlg.m
rename from scripts/java/errordlg.m
rename to scripts/ui/errordlg.m
diff --git a/scripts/java/helpdlg.m b/scripts/ui/helpdlg.m
rename from scripts/java/helpdlg.m
rename to scripts/ui/helpdlg.m
diff --git a/scripts/java/inputdlg.m b/scripts/ui/inputdlg.m
rename from scripts/java/inputdlg.m
rename to scripts/ui/inputdlg.m
diff --git a/scripts/java/listdlg.m b/scripts/ui/listdlg.m
rename from scripts/java/listdlg.m
rename to scripts/ui/listdlg.m
diff --git a/scripts/ui/module.mk b/scripts/ui/module.mk
new file mode 100644
--- /dev/null
+++ b/scripts/ui/module.mk
@@ -0,0 +1,16 @@
+FCN_FILE_DIRS += ui
+
+ui_FCN_FILES = \
+  ui/errordlg.m \
+  ui/helpdlg.m \
+  ui/inputdlg.m \
+  ui/listdlg.m \
+  ui/msgbox.m \
+  ui/questdlg.m \
+  ui/warndlg.m
+
+FCN_FILES += $(ui_FCN_FILES)
+
+PKG_ADD_FILES += ui/PKG_ADD
+
+DIRSTAMP_FILES += ui/$(octave_dirstamp)
diff --git a/scripts/java/msgbox.m b/scripts/ui/msgbox.m
rename from scripts/java/msgbox.m
rename to scripts/ui/msgbox.m
diff --git a/scripts/java/questdlg.m b/scripts/ui/questdlg.m
rename from scripts/java/questdlg.m
rename to scripts/ui/questdlg.m
diff --git a/scripts/java/warndlg.m b/scripts/ui/warndlg.m
rename from scripts/java/warndlg.m
rename to scripts/ui/warndlg.m
